pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -526,38 +543,40 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 		// persistently.
 		//
 		// Claims are treated as "allocated" if they are in the assume cache
-		// or currently their allocation is in-flight.
-		allocator, err := structured.NewAllocator(ctx, allocateClaims, &claimListerForAssumeCache{assumeCache: pl.claimAssumeCache, inFlightAllocations: &pl.inFlightAllocations}, pl.classLister, pl.sliceLister)
+		// or currently their allocation is in-flight. This does not change
+		// during filtering, so we can determine that once.
+		allAllocatedDevices := pl.listAllAllocatedDevices(logger)
+		slices, err := pl.sliceLister.List(labels.Everything())
```

Comment:
The pod would just fail to start if it ended up in a node that no longer satisfies the requirements, right?

Reply:
Yes. We keep discussing how to "bounce it back to the scheduler" because there are other, non-DRA cases where that would be useful, but nothing specific has come out of that yet.

The pod can recover if the device that it needs comes back.